addon/components/course-materials.js

```diff
@@ -1,127 +1,152 @@
-
 import Component from '@ember/component';
-import { Promise as RSVPPromise, map } from 'rsvp';
 import EmberObject, { computed } from '@ember/object';
-import { isPresent, isEmpty } from '@ember/utils';
-import SortableTable from 'ilios-common/mixins/sortable-table';
+import { or, reads } from '@ember/object/computed';
+import { isEmpty, isPresent } from '@ember/utils';
+import { all, task, timeout } from 'ember-concurrency';
 import layout from '../templates/components/course-materials';
 
-export default Component.extend(SortableTable, {
+const DEBOUNCE_DELAY = 250;
+
+export default Component.extend({
```

Comment:
This component is really hard to understand. It was already in pretty bad shape and adding all of these additional things to it made it much more difficult. I think it would be significantly simplified by using only computed properties. Can you spend some time trying that and we can pair on it as well.